UPGRADING.INTERNALS

```diff
@@ -293,6 +293,17 @@ PHP 8.4 INTERNALS UPGRADE NOTES
 4. OpCode changes
 ========================
 
+* DO_ICALL, DO_FCALL, and DO_FCALL_BY_NAME now call zend_interrupt_function
+  while the internal frame is still on the stack. This means interrupt handlers
+  will now see the internal call. If your interrupt handler does something like
+  switching EG(current_execute_data), it probably needs to do something more
+  similar to a fiber switch, or at least not do so when internal frames are on
+  top anymore.
+
+* RETURN, RETURN_BY_REF, GENERATOR_RETURN, and CALL_TRAMPOLINE now do a check
+  for a VM interrupt and call zend_timeout/zend_interrupt_function after the
+  ZEND_OBSERVER_FCALL_END call.
+
```

Comment:
@dstogov Can you check the wording here?

Reply:
I would remove "it probably needs to do something more  similar to a fiber switch" suggestion and keep only the internal frame related restriction.

I didn't agree to add interrupt checks at user function RETURN.